src_c/display.c

```diff
@@ -1011,7 +1011,7 @@ pg_set_mode(PyObject *self, PyObject *arg, PyObject *kwds)
                         if (y == (int)SDL_WINDOWPOS_UNDEFINED_DISPLAY(display))
                             y = SDL_WINDOWPOS_CENTERED_DISPLAY(display);
                     }
-                    else {
+                    else if (!center_window) {
                         SDL_GetWindowPosition(win, &x, &y);
```

Comment:
```suggestion
                    else {
                        int old_w, old_h;
                        SDL_GetWindowSize(win, &old_w, &old_h);

                        /* Emulate SDL1 behaviour: When the window is to be
                         * centred, the window shifts to the new centred
                         * location only when resolution changes and previous
                         * position is retained when the dimensions don't
                         * change.
                         * When the window is not to be centred, previous
                         * position is retained unconditionally */
                        if (!center_window || (w == old_w && h == old_h)) {
                            SDL_GetWindowPosition(win, &x, &y);
                        }
```
I think this should solve the issue I mentioned in my first review.

I tested with the modified version of your example I posted above and this patch does the trick and now the behaviour seems to match 1.9.6 behaviour accurately.

Reply:
Ok now I feel like an idiot, when you said resolution, in my head I stupidly associated it with aspect ratio... That certainly recreates the behaviour